shell/src/main/java/alluxio/cli/LogLevel.java

```diff
@@ -0,0 +1,257 @@
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+
+    List<TargetInfo> targets = parseOptTarget(cmd);
+    String logName = parseOptLogName(cmd);
+    String level = parseOptLevel(cmd);
+
+    for (TargetInfo targetInfo : targets) {
+      if (handleTarget(targetInfo, logName, level)) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  private static List<TargetInfo> parseOptTarget(CommandLine cmd) throws IOException {
+    String[] targets;
+    if (cmd.hasOption(TARGET_OPTION_NAME)) {
+      String argTarget = cmd.getOptionValue(TARGET_OPTION_NAME);
+      if (StringUtils.isBlank(argTarget)) {
+        targets = new String[]{"master", "workers"};
+      } else if (argTarget.contains(TARGET_SEPARATOR)) {
+        targets = argTarget.split(TARGET_SEPARATOR);
+      } else {
+        targets = new String[]{argTarget};
+      }
+    } else {
+      targets = new String[]{"master", "workers"};
+    }
+    return getTargetInfos(targets);
+  }
+
+  private static List<TargetInfo> getTargetInfos(String[] targets) throws IOException {
+    List<TargetInfo> targetInfoList = new ArrayList<>();
+    for (String target : targets) {
+      if (target.equals("master")) {
+        String masterHost = NetworkAddressUtils.getConnectHost(ServiceType.MASTER_WEB);
+        int masterPort = NetworkAddressUtils.getPort(ServiceType.MASTER_WEB);
+        targetInfoList.add(new TargetInfo(masterHost, masterPort, "master"));
+      } else if (target.equals("workers")) {
+        AlluxioBlockStore alluxioBlockStore = AlluxioBlockStore.create();
+        List<BlockWorkerInfo> workerInfoList = alluxioBlockStore.getWorkerInfoList();
+        for (BlockWorkerInfo workerInfo : workerInfoList) {
+          WorkerNetAddress netAddress = workerInfo.getNetAddress();
+          targetInfoList.add(
+              new TargetInfo(netAddress.getHost(), netAddress.getWebPort(), "worker"));
+        }
+      } else if (target.contains(":")) {
+        String[] hostPortPair = target.split(":");
+        int port = Integer.parseInt(hostPortPair[1]);
+        targetInfoList.add(new TargetInfo(hostPortPair[0], port, "worker"));
+      } else {
+        // TODO(maobaolong)
+        // else, throw an exception because the target is unknown
+      }
+    }
+    return targetInfoList;
+  }
+
+  private static String parseOptLogName(CommandLine cmd) {
+    String argName = cmd.getOptionValue(LOG_NAME_OPTION_NAME);
+    if (StringUtils.isNotBlank(argName)) {
+      return argName;
+    }
+    return "";
+  }
+
+  private static String parseOptLevel(CommandLine cmd) {
+    if (cmd.hasOption(LEVEL_OPTION_NAME)) {
+      String argLevel = cmd.getOptionValue(LEVEL_OPTION_NAME);
+      if (StringUtils.isNotBlank(argLevel)) {
+        return argLevel;
+      }
+    }
+    return null;
+  }
+
+  private static boolean handleTarget(TargetInfo targetInfo, String logName, String level) {
```

Comment:
this method would be simpler if we threw an `IOException` instead of returning false on failure